tests/data/diagnosis_summary_json.json

```diff
@@ -74,7 +75,6 @@
         "mem-bw/H2D_Mem_BW:5": null,
         "mem-bw/H2D_Mem_BW:6": null,
         "mem-bw/H2D_Mem_BW:7": null,
-        "mem-bw/return_code": 1.0,
-        "Index": "sb-validation-03"
+        "mem-bw/return_code": 1.0
```

Comment:
will the output file produces float or int?

Reply:
the raw_data_df has been float64 type when it is read from raw dataf file, so I can hardly differentiate the data type of each metric without hardcode

Reply:
seems you call `astype('float64')` explicitly? you may fix it with convert_dtypes or infre_objects in the other pr